src/emit/expression.rs

```diff
@@ -2121,3 +2123,227 @@ fn runtime_cast<'a>(
+                    bin,
+                    &values[i as usize],
+                    to_elem_ty,
+                    vartab,
+                    function,
+                    ns,
+                );
+
+                let output_ptr = unsafe {
+                    bin.builder.build_gep(
+                        llvm_to,
+                        output,
+                        &[i32_const!(i.into()), i32_zero!()],
+                        "output_ptr",
+                    )
+                };
+
+                bin.builder.build_store(output_ptr, ptr);
+
+                let output_len = unsafe {
+                    bin.builder.build_gep(
+                        llvm_to,
+                        output,
+                        &[i32_const!(i.into()), i32_const!(1)],
+                        "output_len",
+                    )
+                };
+
+                bin.builder.build_store(output_len, len);
+            }
+
+            return (output, llvm_length);
+        }
+        Expression::AllocDynamicBytes {
+            initializer: Some(initializer),
+            ..
+        } => {
+            let ptr = bin.emit_global_string("slice_constant", initializer, true);
+            let len = i64_const!(initializer.len() as u64);
+
+            return (ptr, len);
+        }
+        _ => (),
+    }
+
+    let from = e.ty();
+
+    let val = expression(target, bin, e, vartab, function, ns);
+
+    basic_value_to_slice(bin, val, &from, to, function, ns)
+}
+
+/// Convert basic enum value to a slice. This function calls itself recursively
+/// for arrays (become slices of slices).
+fn basic_value_to_slice<'a>(
+    bin: &Binary<'a>,
+    val: BasicValueEnum<'a>,
+    from: &Type,
+    to: &Type,
+    function: FunctionValue<'a>,
+    ns: &Namespace,
+) -> (PointerValue<'a>, IntValue<'a>) {
+    emit_context!(bin);
+
+    match &from {
+        Type::Slice(_) | Type::DynamicBytes | Type::String => {
+            let data = bin.vector_bytes(val);
+            let len = bin.vector_len(val);
+
+            (data, len)
+        }
+        Type::Address(_) => {
+            let address = call!("__malloc", &[i32_const!(ns.address_length as u64).into()])
+                .try_as_basic_value()
+                .left()
+                .unwrap()
+                .into_pointer_value();
+
+            bin.builder.build_store(address, val);
+
+            let len = i64_const!(ns.address_length as u64);
+
+            (address, len)
+        }
+        Type::Bytes(bytes_length) => {
+            let llvm_ty = bin.llvm_type(from, ns);
+            let src = bin.build_alloca(function, llvm_ty, "src");
+
+            bin.builder.build_store(src, val.into_int_value());
+
+            let bytes_length: u64 = (*bytes_length).into();
+
+            let dest = call!("__malloc", &[i32_const!(bytes_length).into()])
+                .try_as_basic_value()
+                .left()
+                .unwrap()
+                .into_pointer_value();
```

Comment:
The other allocation is on the stack and that is a single allocation. This malloc allocation is per-slice.

Reply:
You do not need to allocate space in the stack and in the heap. I can just allocated the memory in one place (heap or stack(, store the integer there, change the endianness in place and return the pointer to that memory region.

Reply:
This might be something for another PR. We do all our endian-ness swaps like this.

In fact, I think `bytesN` should be represented by a byte array and then all these swaps would go away.